sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -459,7 +459,7 @@ export class EventHubSender extends LinkEntity {
           return reject(translate(e));
         };
 
-        const waitTimer = setTimeout(actionAfterTimeout, retryOptions.timeoutInMs);
+        const waitTimer = setTimeout(actionAfterTimeout, retryOptions.timeoutInMs!);
```

Comment:
```suggestion
        const waitTimer = setTimeout(actionAfterTimeout, retryOptions.timeoutInMs ?? 0);
```

Or some other useful default?

Reply:
What about `retryOptions.timeoutInMs || 0` ? Does the same and is easier to understand.

Reply:
I'd prefer to not default here if we can avoid it (ie, let's leave in the !). We have this line:

`retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);` 

At the top of the method to avoid needing to spread defaults throghout.
(I also realize why TS is complaining about it but we can fix that properly later)

Reply:
Leaving the `!` is fine by me, but I was the one who introduced the `!` 😜 at least if you mean the line referenced by this thread specifically.

Reply:
Oh, that's awesome. Yeah, for now your '!' is correct. :)

Reply:
@HarshaNalluru , I feel like you had a PR with timeouts? Perhaps we can fix it there by just declaring a local timeoutInMs.

Reply:
(approving with the assumption you'll figure it out between you and @HarshaNalluru).

Reply:
Unless I'm missing something, omitting the timeout makes `setTimeout` treat it as 0 anyway? https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout

But perhaps what Richard is saying is he'd prefer `retryOptions?.timeoutInMs` which should work since the second argument is optional?

Reply:
@richardpark-msft your turn!

Reply:
@richardpark-msft, my retryTimeout PR was scoped only to service-bus since the update to event-hubs would be a breaking change. 

We don't like `|| 0` since we want to make it clear that the provided timeout is indeed valid just by looking at the code.
The simplest fix to avoid "!" would be to update the line 
`    retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);` 
to 
`    const timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);`
and use the local variable `timeoutInMs` instead of `retryOptions.timeoutInMs` at the lines 462, 472, 504 and 510.

Reply:
Can do! Will do

Reply:
Pushed!

Reply:
Thanks!!